internal/command/local/connect.go

```diff
@@ -203,10 +203,30 @@ func waitConnect(localConfig *config.LocalConnect, out io.Writer) error {
 			goto tick
 
 		}
+		// wait until the local connection has been established
 		connectErrs = sbmgr.CheckStatusConnectErrors(status, ciConfig)
-		if len(connectErrs) == 0 {
-			break
+		if len(connectErrs) != 0 {
+			goto tick
+		}
+		// wait until all local sandboxes are ready (all tunnels have connected)
+		if isRunning, lastError := sbmgr.IsWatcherRunning(status); !isRunning {
+			if lastError == sbmgr.SandboxesWatcherUnimplemented {
+				// this is an old operator, we are done
+				break
+			}
+			goto tick
+		}
+		for i := range status.Sandboxes {
+			sds := status.Sandboxes[i]
+			for j := range sds.LocalWorkloads {
+				lw := sds.LocalWorkloads[j]
```

Comment:
I'd suggest &sds.LocalWorkoads[j] here so that we're not asking for unnecessary copies

Reply:
I did it like that, because `sds.LocalWorkloads` has type `[]*api.SandboxStatus_LocalWorkload`, so both `sds` and `lw` are already pointers.

Reply:
oops, right, sorry I forgot grpc adds pointers everywhere.